### PR TITLE
MCF Browser Switch to Typescript

### PR DESCRIPTION
**Pull request by cwcrystal8**
This PR involves the following changes:

- Resolving lint errors
- Switching the project over to TypeScript
- Fixing the parsing table bug for MCF files

Most of the changes should be formatting changes, type annotations, or type definitions to be used for annotating later.

### CODE REVIEW

File: mcf-browser/src/App.tsx [hunk 1]
```diff
@@ -127,7 +136,8 @@ class App extends Component {
    * param.
    */
   parseUrl() {
-    const params = new URLSearchParams(window.location.hash.trim('#'));
+    console.log('hash: ' + window.location.hash);
```
Comment (beets) [inline mcf-browser/src/App.tsx:139]: please remove console.log

**Review by beets — DISMISSED** [review]
thanks for the update! please remove old js file chunks as new ones gets dded.

File: mcf-browser/src/App.tsx (hunk 1 above)
Comment (cwcrystal8) [reply mcf-browser/src/App.tsx:139]: Removed all instances of console.log throughout the project in the next commit!

**Comment by cwcrystal8** [comment]
ptal, I resolved all merge conflicts and removed all instances of console.log.

**Review by beets — APPROVED** [review]
thanks for the updates!

**Pull request merged**